vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,683 @@
+
+```{r}
+## stringr
+str_to_upper("i") # English
+str_to_upper("i", "tr") # Turkish
+```
+
+```{r,eval=F}
+## base
+Sys.setlocale(category = "LC_ALL", locale = "Turkish") # not run
+```
+
+# Join and split
+
+## `str_c`
+
+*Join multiple strings into a single string*
+
+To join multiple strings into one, base R uses `paste()` while string_r uses `str_c()` (the "c" stands for concatenate).
+
+```{r}
+## stringr
+str_c("Letter: ", letters)
+
+## base
+paste("Letter: ", letters, sep = "")
+```
+
+
+## `str_dup`
+
+*Duplicate and concatenate strings within a character vector*
+
+The stringr package allows for more complicated concatenation of strings with functions such as `str_dup()`. In base R one would have to use a more complicated `paste()` expression to achieve the same result.
+
+```{r}
+## stringr
+fruit <- c("apple", "pear", "banana")
+str_dup(fruit, 2)
+str_dup(fruit, 1:3)
+str_c("ba", str_dup("na", 0:5))
+
+```
+
+## `str_split`
+
+*Split up a string into pieces*
+
+To split a string into pieces with breaks based on a particular pattern match stringr uses `str_split()` and base R uses `strsplit()`. Unlike many of our other examples, when splitting a string into pieces, both base R and stringr use the same default order of inputs (string, then split criteria).
+
+```{r}
+fruits <- c(
+  "apples and oranges and pears and bananas",
+  "pineapples and mangos and guavas"
+)
+
+## stringr
+str_split(fruits, " and ")
+
+## base
+strsplit(fruits, " and ")
+```
+
+The stringr package's `str_split()` allows for more control over the split, including restricting the number of possible matches.
+
+```{r}
+## stringr
+str_split(fruits, " and ", n = 3)
+str_split(fruits, " and ", n = 2)
+
+```
+
+
+
+## `str_glue`
+
+*Format and interpolate a string with glue*
+
+You may want to combine strings and interpolate them at the same time. You can do this with a mix of `paste()` and `eval()` in base R, but stringr makes the process simpler with `str_glue()`.
+
+
+```{r}
+name <- "Fred"
+age <- 50
+anniversary <- as.Date("1991-10-12")
+
+## stringr
+str_glue(
+  "My name is {name}, ",
+  "my age next year is {age + 1}, ",
+  "and my anniversary is {format(anniversary, '%A, %B %d, %Y')}."
+)
+
+## base
+paste("My name is ", eval(name), ", my age next year is ", eval(age + 1), ", and my anniversary is ", eval(format(anniversary, "%A, %B %d, %Y")), ".", sep = "")
+```
+
+```{r}
+## stringr
+mtcars %>% str_glue_data("{rownames(.)} has {hp} hp") %>% head()
+
+## base
+paste(rownames(mtcars), "has", mtcars$hp, "hp") %>% head()
+```
+
+
+# Order strings
+
+## `str_order`
+
+*Order or sort a character vector*
+
+Both base R and stringr have separate functions to order and sort strings.
+
+```{r}
+## stringr
+str_order(letters)
+str_sort(letters)
+
+## base
+order(letters)
+sort(letters)
+```
+
+Some options in `str_order()` and `str_sort()` don't have analogous base R options.
```

Comment:
Again, I don't think you need an example, but you should emphasise that the base functions can return different results for the same inputs.